package.json

```diff
@@ -2,10 +2,17 @@
   "name": "pickdiff",
   "version": "1.0.0",
   "description": "A simple app to selectively diff files between two Git commits.",
-  "main": "server.js",
+  "main": "dist/server.js",
   "scripts": {
-    "start": "node server.js",
-    "dev": "node --watch server.js",
+    "build": "tsc",
+    "build:client": "cd public && npx tsc script.ts --lib DOM,DOM.Iterable,ES2017 --target ES2017 --skipLibCheck",
+    "build:all": "npm run build && npm run build:client",
+    "start": "npm run build && node dist/server.js",
+    "dev": "ts-node server.ts",
+    "dev:watch": "ts-node --watch server.ts",
+    "type-check": "tsc --noEmit",
+    "type-check:client": "cd public && npx tsc script.ts --noEmit --lib DOM,DOM.Iterable,ES2017 --target ES2017 --skipLibCheck",
+    "type-check:all": "npm run type-check && npm run type-check:client",
```

Comment:
Similar comments to the `build` commands above...can this be simplified to a single `type-check` command? Why do we need separate type checking commands/parameters for the client vs server?